### ISSUE TITLE
Dictionary Keycheck on TypedDict not recognized

### ISSUE THREAD

**Issue by Paulchen5** [microsoft/pyright#10662]
**Describe the bug**
Pyright isn't checking that I check the entry of the key in the dictionary

**Code or Screenshots**
If possible, provide a minimal, self-contained code sample (surrounded by triple back ticks) to demonstrate the issue. The code should define or import all referenced symbols.

```python
from typing import TypedDict

class LineInterpretation(TypedDict, total=False):
    indentation: Required[str]
    operator_name: Required[str]
    pre_opId_content: Required[str]
    opId: Required[str]
    result_size: Required[str]
    output_column_size: NotRequired[str]
    rel_addr: Required[str]
    custom_properties: Required[list[dict[str, str]]]

good_case_line: LineInterpretation
bad_case_line: LineInterpretation

if good_case_line.get("output_column_size", -1) != -1 and bad_case_line.get("output_column_size", -1) != -1:
    if good_case_line["output_column_size"] == bad_case_line["output_column_size"]:
```

German pylance error:
```
Auf das Element in TypedDict konnte nicht zugegriffen werden.
  "output_column_size" ist kein erforderlicher Schlüssel in "LineInterpretation". Der Zugriff kann daher zu einer Laufzeitausnahme führen. Pylance[reportTypedDictNotRequiredAccess](https://github.com/microsoft/pylance-release/blob/main/docs/diagnostics/reportTypedDictNotRequiredAccess.md)
```

Translation:
```
The element in TypedDict could not be accessed.
"output_column_size" is not a required key in "LineInterpretation." Accessing it may therefore result in a runtime delay. Pylance [reportTypedDictNotRequiredAccess](https://github.com/microsoft/pylance-release/blob/main/docs/diagnostics/reportTypedDictNotRequiredAccess.md)
```
If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension, a language server in another editor, integrated into Pylance, or the command-line tool? Which version?

Version: `2025.6.2`

**Comment by erictraut** [comment]
If I understand the issue correctly, you are expecting pyright to infer that a particular item is present because you are calling `get` with a second argument (in this case, `-1`) whose type (`int`) does not overlap with the declared type of the item (`str`). 

If I misinterpreted the request, let me know.

That's not a feature that pyright supports, nor do any other Python type checkers. Such a feature could be added, but it seems like it's very specific to your use case and coding style. I haven't seen this pattern used before. The more typical way to write this code is:

```python
def func(good_case_line: LineInterpretation, bad_case_line: LineInterpretation):
    if "output_column_size" in good_case_line and "output_column_size" in bad_case_line:
        if good_case_line["output_column_size"] == bad_case_line["output_column_size"]:
            ...
```

And pyright supports this standard usage pattern.

I'm going to change this from a bug report to an enhancement request, and I'm going to reject the enhancement for now. I'm willing to reconsider the enhancement request in the future if there is signal from enough other pyright users that this usage pattern is something they'd like to see supported.

**Comment by paulchen5** [comment]
Yeah you understand it correct 👍 . Am I allowed to start a survey in the discussions tab? @erictraut

**Comment by erictraut** [comment]
@Paulchen5, we typically use "thumbs up" reactions to gauge interest in feature requests. Others who are interested in seeing this feature can add a reaction to this issue.

**Comment by paulchen5** [comment]
Ok good to know👍

**Comment by erictraut** [comment]
Is there a reason why you don't want to use the more idiomatic (and IMO more readable) way to test for the presence of keys in a dictionary?

**Comment by paulchen5** [comment]
Not really 😅  I didn't thought about it while implementing the code. I have tested both variants and `in` is faster that `.get(…, -1)`. 

I guess `.get(…, -1)` could be helpful if you want to use the result of it afterwards immediately as their wont be an additional memory access I think. It would just be cool to have it implemented too.
